### PR TITLE
[New Rule] busybox shell evasion threat

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/1841

## Summary
busybox an IOC, the Unix binary can be abused to breakout out of restricted shells or environments by spawning an interactive system shell. This activity is not standard use with this binary for a user or system administrator. It indicates a potentially malicious actor attempting to improve the capabilities or stability of their access.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/defense_evasion_busybox_binary.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+references = ["https://gtfobins.github.io/gtfobins/busybox/"]
+risk_score = 47
+rule_id = "e9b4a3c7-24fc-49fd-a00f-9c938031eef1"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and process.name : "busybox" and process.args_count == 2 and process.args : ("/bin/sh", "/bin/ash", "sh", "ash")
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_busybox_binary.toml:27]:
```suggestion
process where event.type == "start" and process.name : "busybox" and process.args_count == 2 and process.args : ("/bin/sh", "/bin/bash", "sh", "bash")
```

File: rules/linux/defense_evasion_busybox_binary.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+references = ["https://gtfobins.github.io/gtfobins/busybox/"]
+risk_score = 47
+rule_id = "e9b4a3c7-24fc-49fd-a00f-9c938031eef1"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and process.name : "busybox" and process.args_count == 2 and process.args : ("/bin/sh", "/bin/bash", "sh", "bash")
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_busybox_binary.toml:27]:
```suggestion
process where event.type == "start" and process.name : "busybox" and process.args_count == 2 and process.args : ("/bin/sh", "/bin/ash", "sh", "ash")
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/defense_evasion_busybox_binary.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+creation_date = "2022/03/15"
+maturity = "production"
+updated_date = "2022/03/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies Linux binary busybox abuse to break out from restricted environments by spawning an interactive system
+shell.This activity is not standard use with this binary for a user or system administrator. It indicates a potentially
+malicious actor attempting to improve the capabilities or stability of their access.
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_busybox_binary.toml:11]: Maybe we can elaborate a less generic description, describe what the binary does, etc

File: rules/linux/defense_evasion_busybox_binary.toml (hunk 1 above)
Comment (w0rk3r) [inline rules/linux/defense_evasion_busybox_binary.toml:27]: Maybe do you meant `bash` on this one?

Comment (w0rk3r) [reply rules/linux/defense_evasion_busybox_binary.toml:27]: Also, is args_count needed here?

File: rules/linux/defense_evasion_busybox_binary.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1548"
+name = "Abuse Elevation Control Mechanism"
+reference = "https://attack.mitre.org/techniques/T1548/"
+[[rule.threat.technique.subtechnique]]
+id = "T1548.004"
+name = "Elevated Execution with Prompt"
+reference = "https://attack.mitre.org/techniques/T1548/004/"
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_busybox_binary.toml:40]:
```suggestion
```
I don't think this technique fits here, if the execution of the command does not prompt the user for credentials

**Review by w0rk3r — CHANGES_REQUESTED** [review]

File: rules/linux/defense_evasion_busybox_binary.toml (hunk 4 above)
Comment (Samirbous) [reply rules/linux/defense_evasion_busybox_binary.toml:40]: ++ https://attack.mitre.org/techniques/T1059/004/

**Review by Samirbous — APPROVED** [review]

File: rules/linux/defense_evasion_busybox_binary.toml (hunk 3 above)
Comment (shashank-elastic) [reply rules/linux/defense_evasion_busybox_binary.toml:11]: Response in https://github.com/elastic/detection-rules/pull/1840

File: rules/linux/defense_evasion_busybox_binary.toml (hunk 1 above)
Comment (shashank-elastic) [reply rules/linux/defense_evasion_busybox_binary.toml:27]: Its definetly ash :) 
image

Comment (shashank-elastic) [reply rules/linux/defense_evasion_busybox_binary.toml:27]: args_count tightens the query and avoids false positives like below and saves the user from analyzing the alert 
```
{
  "_id": "-cJzjn8BfJgn-82XzsUc",
  "_index": ".ds-logs-endpoint.events.process-default-2022.03.09-000005",
  "_score": "1",
  "_type": "_doc",
  "@timestamp": "2022-03-15T16:40:00.536Z",
  "agent": {
    "id": "c8a71895-37f3-48c2-ba8e-8bf9e844ac1d",
    "type": "endpoint",
    "version": "7.15.1"
  },
  "data_stream": {
    "dataset": "endpoint.events.process",
    "namespace": "default",
    "type": "logs"
  },
  "ecs": {
    "version": "1.11.0"
  },
  "elastic": {
    "agent": {
      "id": "c8a71895-37f3-48c2-ba8e-8bf9e844ac1d"
    }
  },
  "event": {
    "action": "exec",
    "agent_id_status": "verified",
    "category": "process",
    "created": "2022-03-15T16:40:00.536Z",
    "dataset": "endpoint.events.process",
    "id": "MUGdZk4JkGfMCVNH+++K1WBT",
    "ingested": "2022-03-15T16:40:28.000Z",
    "kind": "event",
    "module": "endpoint",
    "sequence": "660514",
    "type": "start"
  },
  "group": {
    "Ext": {
      "real": {
        "id": "1001",
        "name": "shashankks"
      }
    },
    "id": "1001",
    "name": "shashankks"
  },
  "host": {
    "architecture": "x86_64",
    "hostname": "elastic-test-instance-3",
    "id": "57d2b7239218c2b7728faa7cfdf6fd2f",
    "ip": "127.0.0.1,::1,10.128.0.19,fe80::4001:aff:fe80:13",
    "mac": "42:01:0a:80:00:13",
    "name": "elastic-test-instance-3",
    "os": {
      "Ext": {
        "variant": "Debian"
      },
      "family": "debian",
      "full": "Debian 10",
      "kernel": "4.19.0-18-cloud-amd64 #1 SMP Debian 4.19.208-1 (2021-09-29)",
      "name": "Linux",
      "platform": "debian",
      "type": "linux",
      "version": "10"
    }
  },
  "message": "Endpoint process event",
  "process": {
    "args": "busybox,ash,touch",
    "args_count": "3",
    "command_line": "busybox ash touch",
    "entity_id": "YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU1MzktMTMyOTE4MzYwMDAuNTM2MDk5MTAw",
    "executable": "/usr/bin/busybox",
    "Ext": {
      "ancestry": "YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU1MzktMTMyOTE4MzYwMDAuNTMzODIwMzAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU0NTQtMTMyOTE4Mjk5NjYuODI4MTk2MTAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU0NTQtMTMyOTE4Mjk5NjYuNzg1MTU3NzAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU0NTMtMTMyOTE4Mjk5NjYuMTUwNDc3MDAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU0NDctMTMyOTE4Mjk5NjQuMzgzODEzODAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU0NDctMTMyOTE4Mjk5NjQuMjY4ODI2NjAw,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTUxOC0xMzI4ODk0NjA2Ny4w,YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTEtMTMyODg5NDYwNjIuMA=="
    },
    "hash": {
      "md5": "93e29acf3b67fe8847105371d3bd5f80",
      "sha1": "062c7c785e1c87800c6b76d6648cb10e2297f370",
      "sha256": "f030c248efbe4e2337aac9d53726e9222afd81dc802ad04ee1092dc92d1d15f3"
    },
    "name": "busybox",
    "parent": {
      "args": "-bash",
      "args_count": "0",
      "command_line": "-bash",
      "entity_id": "YzhhNzE4OTUtMzdmMy00OGMyLWJhOGUtOGJmOWU4NDRhYzFkLTU1MzktMTMyOTE4MzYwMDAuNTMzODIwMzAw",
      "executable": "/bin/bash",
      "name": "bash",
      "pid": "5454"
    },
    "pid": "5539"
  },
  "user": {
    "Ext": {
      "real": {
        "id": "1000",
        "name": "shashankks"
      }
    },
    "id": "1000",
    "name": "shashankks"
  }
}```

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**